app/models/spree/payment.rb

```diff
@@ -77,6 +78,15 @@ class Payment < ApplicationRecord
       event :invalidate do
         transition from: [:checkout], to: :invalid
       end
+      event :require_authorization do
+        transition from: [:checkout, :processing], to: :requires_authorization
+      end
+      event :failed_authorization do
+        transition from: [:requires_authorization], to: :failed
+      end
+      event :completed_authorization do
```

Comment:
this doesn't feel consistent with `require_authorization`. While this one is a participle the other one is an imperative verb. I'd stick to one or the other. Perhaps an imperative makes more sense when reading like `payment.complete_authorization`?

Reply:
Agreed. I addressed this and `failed_authorization` in 2d7676d87f852966446b0b172cbd222580ab9b6c. It would be nice to also change `started_processing` to `start_processing` to be totally consistent, but I want to keep this PR small :)